webapp/src/js/components/Plot.js

```diff
@@ -32,7 +30,24 @@ let Plot = createReactClass({
     };
   },
 
+  componentWillMount() {
+    Promise.all([
+      import('plotly.js/dist/plotly-cartesian'),
+      //...more imports
+    ]).then(([Plotly]) => {
+      this.hasImported = true;
+      this.Plotly = Plotly;
+      this.PlotlyComponent = createPlotlyComponent(this.Plotly);
+      this.forceUpdate();
+    }).catch((error) => 'An error occurred while loading the component: ' + error);
+  },
+
   render() {
+
+    if (!this.hasImported) {
```

Comment:
We could render `<Loading>` here instead of being blank.

Reply:
Cool, nice idea.